src/cards/vacuum-card/controls/vacuum-buttons-control.ts

```diff
@@ -0,0 +1,131 @@
+    isStopped,
+    isUnavailable,
+    supportsCleanSpot,
+    supportsLocate,
+    supportsPause,
+    supportsReturnHome,
+    supportsStart,
+    supportsStop,
+} from "../utils";
+
+interface VacuumCommand {
+    icon: string;
+    serviceName: string;
+    isVisible: (entity: HassEntity) => boolean;
+    isDisabled: (entity: HassEntity) => boolean;
+}
+
+const VACUUM_COMMANDS: VacuumCommand[] = [
+    {
+        icon: "play",
+        serviceName: "start",
+        isVisible: (entity) => supportsStart(entity),
+        isDisabled: (entity) => isUnavailable(entity) || isCleaning(entity),
+    },
+    {
+        icon: "pause",
+        serviceName: "pause",
+        isVisible: (entity) => supportsStart(entity) && supportsPause(entity),
+        isDisabled: (entity) => isUnavailable(entity) || !isCleaning(entity),
+    },
+    {
+        icon: "play-pause",
+        serviceName: "start_pause",
+        isVisible: (entity) => !supportsStart(entity) && supportsPause(entity),
+        isDisabled: (entity) => isUnavailable(entity),
+    },
+    {
+        icon: "stop",
+        serviceName: "stop",
+        isVisible: (entity) => supportsStop(entity),
+        isDisabled: (entity) => isUnavailable(entity) || isStopped(entity),
+    },
+    {
+        icon: "target-variant",
+        serviceName: "clean_spot",
+        isVisible: (entity) => supportsCleanSpot(entity),
+        isDisabled: (entity) => isUnavailable(entity),
+    },
+    {
+        icon: "home-map-marker",
+        serviceName: "return_to_base",
+        isVisible: (entity) => supportsReturnHome(entity),
+        isDisabled: (entity) => isUnavailable(entity) || isReturningHome(entity),
+    },
+    {
+        icon: "map-marker",
+        serviceName: "locate",
+        isVisible: (entity) => supportsLocate(entity),
+        isDisabled: (entity) => isUnavailable(entity),
+    },
+];
+
+@customElement("mushroom-vacuum-buttons-control")
+export class VacuumButtonsControl extends LitElement {
+    @property({ attribute: false }) public hass!: HomeAssistant;
+
+    @property({ attribute: false }) public entity!: HassEntity;
+
+    @property() public fill: boolean = false;
+
+    private _callService(e: MouseEvent, serviceName: string): void {
+        e.stopPropagation();
+        this.hass.callService("vacuum", serviceName, {
+            entity_id: this.entity.entity_id,
+        });
+    }
+
+    protected render(): TemplateResult {
+        return html`
+            <div
+                class=${classMap({
+                    container: true,
+                    fill: this.fill,
+                })}
+            >
+                ${VACUUM_COMMANDS.filter((item) => item.isVisible(this.entity))
+                    .slice(0, 4)
+                    .map(
+                        (item) => html`
+                            <mushroom-button
+                                .icon="mdi:${item.icon}"
+                                .disabled=${item.isDisabled(this.entity)}
```

Comment:
```suggestion
                                .disabled=${isUnavailable(this.entity) || item.isDisabled(this.entity)}
```
Do avoid repeating this condition in all item commands.